### PR TITLE
[XPU] gaussian_random support fp16

### PR DESCRIPTION
**Pull request by shentanyue**
### PR types

New features
### PR changes

OPs 
### Describe

xpu 的 gaussian_random op 支持 fp16

### CODE REVIEW

File: paddle/phi/kernels/xpu/gaussian_kernel.cc [hunk 1]
```diff
@@ -29,7 +29,7 @@ void GaussianKernel(const Context& ctx,
                     int seed,
                     DataType dtype,
                     DenseTensor* out) {
-  std::normal_distribution<T> dist(mean, std);
+  std::normal_distribution<float> dist(mean, std);
```
Comment (shentanyue) [inline paddle/phi/kernels/xpu/gaussian_kernel.cc:32]: std::normal_distribution's inputs must be a floating point type.

**Review by qili93 — APPROVED** [review]
LGTM

**Review by RuohengMa — APPROVED** [review]
LGTM

**Review by ykkk2333 — APPROVED** [review]
LGTM

**Pull request merged**